From a48460b41399112f5d3ae03b30e3f9e885346ff7 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 21 Sep 2015 15:37:08 +0800
Subject: [PATCH] MLK-11594 video: mxsfb: correct the pm runtime logic

The pm power's usage_count should be first added by
one before putting it into suspend state. Otherwise
the usage_count will be negative and the power cannot
be put into suspend state correctly.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index 25bdf31..4d38ef3 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -896,9 +896,7 @@ static int mxsfb_blank(int blank, struct fb_info *fb_info)
 	case FB_BLANK_NORMAL:
 		if (host->enabled) {
 			mxsfb_disable_controller(fb_info);
-			pm_runtime_put_noidle(&host->pdev->dev);
 			pm_runtime_put_sync_suspend(&host->pdev->dev);
-			pm_runtime_get_noresume(&host->pdev->dev);
 		}
 
 		clk_disable_disp_axi(host);
@@ -1478,6 +1476,7 @@ static int mxsfb_probe(struct platform_device *pdev)
 		writel(0, host->base + LCDC_CTRL);
 		mxsfb_set_par(fb_info);
 		mxsfb_enable_controller(fb_info);
+		pm_runtime_get_sync(&host->pdev->dev);
 	}
 
 	ret = register_framebuffer(fb_info);
-- 
1.8.0

